### ISSUE TITLE
Bug: Player Ghost Monster Recall

### ISSUE THREAD

**Issue by ThatGitMITZE** [NickMcConnell/FAangband#293]
Not entirely sure what happened, but I have ", the Mummy" permanently (or possibly just until I find another Mummy player ghost and kill it) stuck in my monster recall; I suspect what happened is that if you leave the player ghost alive when you leave the level, it's not removed from the monster recall properly—though I don't have any good proof of this other than a gut feeling that that's what happened. I suppose it's also possible it's specifically related to Mummy player ghosts.

**Comment by NickMcConnell** [comment]
This is very like #279.

**Comment by ThatGitMITZE** [comment]
More information on this:

Decided to go try bugtest it on the webserver; I had a ", the Mummy" in my monster recall when I started.

Upon encountering my first player ghost (MITZE II, the Skeleton), the ", the Mummy" entry changed to "MITZE II, the Mummy" despite not being the player ghost encountered.

After leaving the level without killing "MITZE II, the Skeleton" that entry disappeared from the monster memory and the "MITZE II, the Mummy" entry changed back to ", the Mummy".

After generating a second player ghost (also "MITZE II, the Skeleton") I then died on the level without killing it; on starting a new game, there was now a ", the Skeleton" in the monster memory in addition to one for ", the Mummy".

tl;dr I think the issue is somewhere either with getting killed by a player ghost, or more probably with just dying while a player ghost is generated on the level somewhere. I guess something about dying doesn't clear the requisite parts of monster memory while just leaving the level does?
